### PR TITLE
[Promotion] Product filter

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Related tickets | continuation for #6480 
| License         | MIT

### CODE REVIEW

File: features/promotion/receiving_discount/receiving_fixed_discount_on_specific_products.feature [hunk 1]
```diff
@@ -0,0 +1,64 @@
+@receiving_discount
+Feature: Receiving fixed discount on specific products
+    In order to pay less while buying promoted goods
+    As a Customer
+    I want to receive discount for my purchase
```
Comment (michalmarcinkowski) [inline features/promotion/receiving_discount/receiving_fixed_discount_on_specific_products.feature]: I want to receive discount on each unit of promoted product

File: features/promotion/receiving_discount/receiving_fixed_discount_on_specific_products.feature [hunk 2]
```diff
@@ -0,0 +1,64 @@
+@receiving_discount
+Feature: Receiving fixed discount on specific products
+    In order to pay less while buying promoted goods
```
Comment (michalmarcinkowski) [inline features/promotion/receiving_discount/receiving_fixed_discount_on_specific_products.feature]: In order to buy specific product with a discount

File: src/Sylius/Behat/Context/Setup/PromotionContext.php [hunk 3]
```diff
@@ -11,6 +11,7 @@
 
 namespace Sylius\Behat\Context\Setup;
 
+use Behat\Behat\Tester\Exception\PendingException;
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Setup/PromotionContext.php]: Redundant.

File: src/Sylius/Behat/Context/Setup/PromotionContext.php [hunk 4]
```diff
@@ -564,6 +565,22 @@ public function thePromotionGivesOffIfOrderContainsProducts(PromotionInterface $
...
+    {
+        $this->createUnitFixedPromotion($promotion, $discount, $this->getProductsFilterConfiguration([$product->getCode()]));
+    }
+
+    /**
+     * @Given /^([^"]+) gives ("[^"]+%") off on a ("[^"]*" product)$/
+     */
+    public function itGivesPercentageDiscountOffOnAProduct(PromotionInterface $promotion, $discount, ProductInterface $product)
+    {
+        $this->createUnitPercentagePromotion($promotion, $discount, $this->getProductsFilterConfiguration([$product->getCode()]));
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Setup/PromotionContext.php]: This promotion will be applied to every order, but not every order will be affected by it's action. We should only apply the promotion to orders that contain this product, so we should add a promotion rule `ContainsProduct` and probably modify the step to "it gives "20%" off on a "PHP T-Shirt" product **if order contains it**"

Comment (lchrusciel) [inline src/Sylius/Behat/Context/Setup/PromotionContext.php]: But this is not an idea. We have `ContainsProduct` rule already, but this step should create a promotion that will be applied to every order, but only affect specified items.

Comment (lchrusciel) [inline src/Sylius/Behat/Context/Setup/PromotionContext.php]: But thats a point of this test. This promotion should be assigned to every order, but not every item should be affected. The following scenario checks if filter works properly:
```yml
    Background:
        Given the store operates on a single channel in "United States"
        And the store has a product "PHP T-Shirt" priced at "$100.00"
        And the store has a product "PHP Mug" priced at "$20.00"
        And there is a promotion "T-Shirts promotion"
        And it gives "$10.00" off on a "PHP T-Shirt" product

    @ui
    Scenario: Receiving fixed discount on a single item
        When I add product "PHP T-Shirt" to the cart
        Then its price should be decreased by "$10.00"
        And my cart total should be "$90.00"

    @ui
    Scenario: Not receiving fixed discount on another item
        When I add product "PHP Mug" to the cart
        Then product "PHP Mug" price should not be decreased
        And my cart total should be "$20.00"
```

**Comment by pjedrzejewski** [comment]
Merged in #6489.

**Comment by pjedrzejewski** [comment]
Thank you Łukasz!

**Pull request merged**